FullStoryMiddleware/FullStoryMiddleware.m

```diff
@@ -117,15 +130,25 @@ - (SEGPayload *)getNewPayloadWithFSURL:(SEGContext * _Nonnull)context {
             NSMutableDictionary *newProps = [[NSMutableDictionary alloc] initWithDictionary:screenPayload.properties];
             [newProps setValue:[FS currentSessionURL] forKey:@"FSSessionURL"];
 
-            // SEGScreenPayload initWithName signiture has changed @v4.1
-            // temporary workaround to set the catagory for post 4.1
-            SEGScreenPayload *screen = [[SEGScreenPayload alloc] initWithContext:screenPayload.context integrations:screenPayload.integrations];
-            [screen setValue:[screenPayload.name copy] forKey:@"name"];
-            [screen setValue:[newProps copy] forKey:@"properties"];
-            if(screen.category) {
-                [screen setValue:[screenPayload.category copy] forKey:@"category"];
+            {
+                SEGScreenPayload *uninitPayload = [SEGScreenPayload alloc];
+
+                // init has a different signiture for Analytics version < 4
+                if ([uninitPayload respondsToSelector:@selector(initWithName:properties:context:integrations:)]) {
+                    newPayload = [uninitPayload
+                               initWithName:screenPayload.name
+                               properties:newProps
+                               context:screenPayload.context
+                               integrations:screenPayload.integrations];
+                }else if ([uninitPayload respondsToSelector:@selector(initWithName:category:properties:context:integrations:)]) {
+                    newPayload = [uninitPayload
+                               initWithName:screenPayload.name
+                               category:screenPayload.category
+                               properties:newProps
+                               context:screenPayload.context
+                               integrations:screenPayload.integrations];
+                }
```

Comment:
If it responds to neither selector, then we're left with an uninitialized value. This case should be handled somehow.

---

Also, Objective-C selector calls are conventionally vertically-aligned by `:`. I don't think the convention is pretty but it's standard, and Xcode usually does this for you if you hit tab before the argument label.

```suggestion
                // init has a different signiture for Analytics version < 4
                if ([uninitPayload respondsToSelector:@selector(initWithName:properties:context:integrations:)]) {
                    newPayload = [uninitPayload initWithName:screenPayload.name
                                                  properties:newProps
                                                     context:screenPayload.context
                                                integrations:screenPayload.integrations];
                }else if ([uninitPayload respondsToSelector:@selector(initWithName:category:properties:context:integrations:)]) {
                    newPayload = [uninitPayload initWithName:screenPayload.name
                                                    category:screenPayload.category
                                                  properties:newProps
                                                     context:screenPayload.context
                                                integrations:screenPayload.integrations];
                }
```

Reply:
Good point! added else to set newPayload to nil, which is then handled downstream